go/vt/vtgate/semantics/early_rewriter.go

```diff
@@ -140,12 +149,18 @@ func rewriteHavingAndOrderBy(cursor *sqlparser.Cursor, node sqlparser.SQLNode) {
 			if !col.Qualifier.IsEmpty() {
 				return false
 			}
+			_, parentIsAggr := inner.Parent().(sqlparser.AggrFunc)
```

Comment:
Just skimmed this PR and noticed something:

While this fixes the case of simple aggregations like `sum(foo)` the new code still does not handle more _complex_ aggregations correctly.

For example

```sql
select id, sum(foo) as foo from t1 having sum(foo + 1) > 1
```

This is still rewritten to

```sql
select id, sum(foo) as foo from t1 having sum(sum(foo) + 1) > 1
```